libraries/libpensol/src/main/java/org/pentaho/reporting/libraries/pensol/PublishRestUtil.java

```diff
@@ -109,32 +114,30 @@ public int publishFile( String filePath, byte[] data, boolean overwriteIfExists
    *
    * @param repositoryPath    repository folder where to upload the file; must be separated with /
    * @param fileName          uploaded file's name; must not contain prohibited symbols
+   * @param fileProperties    the properties which should be applied to the file on repository
    * @param fileInputStream   file's data stream
-   * @param overwriteIfExists flag indicating an existing file should be overwritten
    * @return http response code
    * @see org.pentaho.platform.web.http.api.resources.RepositoryPublishResource
    */
-  public int publishFile( String repositoryPath, String fileName, InputStream fileInputStream,
-                          boolean overwriteIfExists ) throws IOException {
-
-    String url = baseUrl + REPO_FILES_IMPORT;
-
+  public int publishFile( String repositoryPath, String fileName, InputStream fileInputStream, Optional<Properties> fileProperties ) throws IOException {
+    String url = baseUrl + REPO_FILES_IMPORT_WITH_OPTIONS;
     WebResource resource = client.resource( url );
     int responseCode = 504;
-    try {
-      FormDataMultiPart part = new FormDataMultiPart();
+    try ( ByteArrayOutputStream baos = new ByteArrayOutputStream() ) {
+      FormDataMultiPart fdmp = new FormDataMultiPart();
 
       String pathEncoded = URLEncoder.encode( repositoryPath + "/" + fileName, "UTF-8" );
       String nameEncoded = URLEncoder.encode( fileName, "UTF-8" );
 
-      part.field( "importPath", pathEncoded, MULTIPART_FORM_DATA_TYPE );
-      part.field( "fileUpload", fileInputStream, MULTIPART_FORM_DATA_TYPE );
-      part.field( "overwriteFile", Boolean.toString( overwriteIfExists ), MULTIPART_FORM_DATA_TYPE );
-      part.getField( "fileUpload" )
-        .setContentDisposition( FormDataContentDisposition.name( "fileUpload" ).fileName( nameEncoded ).build() );
+      fileProperties.orElseGet( () -> new Properties() ).storeToXML( baos, "file properties", "UTF-8" );
+
+      fdmp.bodyPart( new FormDataBodyPart( "properties", baos.toString( "UTF-8"  ), APPLICATION_XML_TYPE ) );
```

Comment:
You are missing the "overwriteFile" flag here. This means republishing over an existing report will fail. Please restore that line.